deploy/crds/rhjmc.redhat.com_flightrecorders_crd.yaml

```diff
@@ -368,10 +368,36 @@ spec:
                     description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
                     type: string
                 type: object
+              templates:
+                description: Listing of templates available in the target JVM
+                items:
+                  description: TemplateInfo contains metadata for a JFR template
+                  properties:
+                    description:
+                      description: A description of the template and its performance
+                        impact
+                      type: string
+                    name:
+                      description: The name of the template
+                      type: string
+                    provider:
+                      description: The organization which has provided the template
+                      type: string
+                    type:
+                      description: The type of template
```

Comment:
The `type` field only has two possible values, `TARGET` and `CUSTOM`. These denote that the template either comes from the target JVM as a built-in preset template or that it was added to ContainerJFR (probably be the end user). I think this is worth documenting here, since the values are briefly enumerable and the meanings are well-defined.

Reply:
In that case, we could make `type` an enum type like we do for `RecordingState` here:
https://github.com/rh-jmc-team/container-jfr-operator/blob/796a4f931f66b01bca513a0f33fc86b878a0d17e/pkg/apis/rhjmc/v1beta1/recording_types.go#L69-L86

Then we can also add OpenAPI validation on the field to enforce these values, like this:
https://github.com/rh-jmc-team/container-jfr-operator/blob/796a4f931f66b01bca513a0f33fc86b878a0d17e/pkg/apis/rhjmc/v1beta1/recording_types.go#L91-L94